collector/sources.go

```diff
@@ -121,6 +171,15 @@ func (e Exporter) getSourcesMetrics(logger log.Logger, ch chan<- prometheus.Metr
 		sourceAddress := r.IPAddr.String()
 		sourceName := e.dnsLookup(logger, r.IPAddr)
 
+		ntpDataPacket, err := client.Communicate(chrony.NewNTPDataPacket(r.IPAddr))
```

Comment:
Hmm, I'm wondering if this should be a separate collector. This adds a new communication for each source.

Reply:
That was actually my first attempt. However, the NTPDataPacket function needs to be called for each source/peer with the respective peer ip address as parameter. A separate collector would look quite similar to the source collector.

But if you prefer I can also implement this separately.

Reply:
The other option would be to add a flag to enable this additional data collection.

Reply:
I have pushed a new version which adds a flag (and defaults to "off"/false).